packages/jsii-pacmak/lib/targets/golang/types/class.ts

```diff
@@ -121,52 +130,29 @@ export class ClassMethod implements TypeField {
   /* emit generates method on the class */
   public emit(code: CodeMaker) {
     const name = code.toPascalCase(this.method.name);
-    const returnType = `${this.returnType ? `${this.returnType} ` : ''}`;
+    const returnType = `${
+      this.returnTypeString ? `${this.returnTypeString} ` : ''
+    }`;
     const instanceArg = this.parent.name.substring(0, 1).toLowerCase();
 
-    let params: string[] = [];
-    if (this.method.parameters) {
-      params = this.method.parameters.map(
-        // TODO: resolve types for parameters
-        (x) => `"${x.name} ${x.type.toString()}"`,
-      );
-    }
-    const parameters = params.length === 0 ? '' : params.join(', ');
-
     code.openBlock(
-      `func (${instanceArg} *${this.parent.name}) ${name}() ${returnType}`,
+      `func (${instanceArg} *${code.toPascalCase(
+        this.parent.name,
+      )}) ${name}() ${returnType}`,
     );
 
-    // FIXME Placeholder
-    code.open(`jsii.NoOpRequest(jsii.NoOpApiRequest {`);
-    code.line(`Class: "${this.parent.name}",`);
-    code.line(`Method: "${name}",`);
-    code.line(`Parameters: []string{${parameters}}`);
-    code.close(`})`);
-
-    const ret = this.references;
-    if (ret?.type?.type.isClassType()) {
-      code.line(`return ${this.returnType}{}`);
-    } else if (ret?.type?.type.isEnumType()) {
-      code.line(`return "ENUM_DUMMY"`);
-    } else {
-      code.line(`return ${this.getDummyReturn(this.returnType)}`);
-    }
+    this.runtimeCall.emit(code);
 
     code.closeBlock();
     code.line();
   }
 
   public emitSignature(code: CodeMaker) {
```

Comment:
Nit: I know this isn't part of the original change, but for consistency maybe we can call this `emitDecl`? See: https://github.com/aws/jsii/blob/master/packages/jsii-pacmak/lib/targets/golang/types/class.ts#L38-L43